dashboard/src/components/map/popup/MapPopupContent.tsx

```diff
@@ -0,0 +1,120 @@
+'use client';
+
+import { FlagIconProps } from '@/components/icons';
+import { CountryDisplay } from '@/components/language/CountryDisplay';
+import { TrendPercentage } from '@/components/TrendPercentage';
+import { SupportedLanguages } from '@/constants/i18n';
+import type { GeoVisitorWithCompare } from '@/entities/geography';
+import { TimeRangeContextProps } from '@/contexts/TimeRangeContextProvider';
+import { cn } from '@/lib/utils';
+import { getCountryName } from '@/utils/countryCodes';
+import { formatPrimaryRangeLabel } from '@/utils/formatPrimaryRangeLabel';
+import { formatNumber } from '@/utils/formatters';
+import { useTranslations } from 'next-intl';
+import React from 'react';
+
+export type MapPopupContentProps = {
+  geoVisitor?: GeoVisitorWithCompare;
+  className?: string;
+  locale: SupportedLanguages;
+  size?: 'sm' | 'lg';
+  t: ReturnType<typeof useTranslations<'components'>>;
+  timeRangeCtx: TimeRangeContextProps;
+  onMouseEnter?: () => void;
+};
+
+function MapPopupContentComponent({
+  geoVisitor,
+  locale,
+  className,
+  size = 'sm',
+  t,
+  timeRangeCtx,
+  onMouseEnter,
+}: MapPopupContentProps) {
+  if (!geoVisitor) return null;
+
+  const hasCmp = !!geoVisitor.compare?.compareVisitors;
+
+  const Row = ({
+    color,
+    label,
+    value,
+    muted,
+    tooltip,
+  }: {
+    color?: string;
+    label: React.ReactNode;
+    value: React.ReactNode;
+    tooltip?: string;
+    muted?: boolean;
+  }) => (
+    <div className='flex items-center justify-between gap-2'>
+      <div className='flex items-center gap-2' title={tooltip}>
+        {color && <div className={`${color} h-2 w-2 rounded-full`} />}
+        <span className={muted ? 'text-muted-foreground' : ''}>{label}</span>
+      </div>
+      <span className={cn('font-medium', muted && 'text-muted-foreground')}>{value}</span>
```

Comment:
Quibble: Use `cn` for all the `classNames`
```suggestion
        {color && <div className={cn(color, 'h-2 w-2 rounded-full')} />}
        <span className={cn({ 'text-muted-foreground': muted })}>{label}</span>
      </div>
      <span className={cn('font-medium', muted && 'text-muted-foreground')}>{value}</span>
```